File: client/src/models/Cult.js
import { DatabaseItem } from "./DatabaseItem.js";
import { Profile } from "./Profile.js";

export class Cult extends DatabaseItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.coverImg = data.coverImg
    this.leaderId = data.leaderId
    this.leader = new Profile(data.leader)
  }
}

File: server/Models/Cult.cs
namespace winter24_insta_cult.Models;

public class Cult : RepoItem<int> // NOTE inherits all members from RepoItem, and we supply a type for the Id
{
  public string Name { get; set; }
  public string Description { get; set; }
  public string CoverImg { get; set; }
  public string LeaderId { get; set; }
  public Profile Leader { get; set; }
}